connmgr/connmanager.go

```diff
@@ -530,8 +525,8 @@ func (cm *ConnManager) Connect(ctx context.Context, c *ConnReq) {
 		return
 	}
 
-	if atomic.LoadUint64(&c.id) == 0 {
-		atomic.StoreUint64(&c.id, atomic.AddUint64(&cm.connReqCount, 1))
+	if c.id.Load() == 0 {
+		c.id.Store(cm.connReqCount.Add(1))
```

Comment:
not the fault of your patch, but this looks very racy.  c.id may not longer be 0 by the time we store the value.

Reply:
I don't see a straightforward way to implement this correctly using atomics.  Both c.id and cm.connReqCount need to be updated atomically together.  This might require a mutex instead.

Reply:
Moved to #3054